server/users/service.go

```diff
@@ -211,7 +219,11 @@ func (rs *RecognitionService) runPipeline(destinationID string, identifiers Even
 	}
 
 	for storedEventID, storedSerializedEvent := range eventsMap {
-		event := map[string]interface{}{}
+		if storedEventID == identifiers.EventID {
```

Comment:
Do we really need this check?

Reply:
In current behaviour, it can be removed.